{
  try {
    runnable.run();
    authenticationSuccess.update(1);
  }
 catch (  PrestoException e) {
    authenticationFail.update(1);
    throw e;
  }
}
